src/Form/LinkField.php

```diff
@@ -60,10 +62,18 @@ public function saveInto(DataObjectInterface $record)
         return $this;
     }
 
+    public function getSchemaStateDefaults()
+    {
+        $data = parent::getSchemaStateDefaults();
+        $data['canCreate'] = $this->getOwner()->canEdit() && Link::singleton()->canCreate();
```

Comment:
```suggestion
        $data['canCreate'] = $this->getOwner()->canEdit();
```
You made a good point while we were discussing this in person - we have to respect the actual type of link being created - if we can create a `PhoneLink` but not `Link` then we need to respect that. You're doing that with the change to `AllowedLinkClassesTrait` so we should remove the `canCreate()` check from here.

You can also remove the `canEdit` (and this entire prop and its functionality) if you want - if owner's `canEdit()` method returns `false` then  the field will be set to readonly (which is being handled separately in https://github.com/silverstripe/silverstripe-linkfield/issues/11) so this prop won't matter in that scenario anyway.

We should probably also add some logic (assuming it's not there already) so that when there are _no_ available link types for the field, it has a nice message instead of a weirdly empty link type menu - though I'd be okay with that being spun off separately.

Reply:
Please also change this in the other places that pass this prop.

Reply:
Updated

Have added AC to the [readonly card](https://github.com/silverstripe/silverstripe-linkfield/issues/11) to revisit the canEdit() logic, decide what to do with it then

Have created [new card](https://github.com/silverstripe/.github/issues/177) to handle the empty list scenario